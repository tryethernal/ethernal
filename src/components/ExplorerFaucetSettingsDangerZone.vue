<template>
    <v-sheet outlined color="error" rounded>
        <Explorer-Faucet-Private-Key-Export-Modal ref="explorerFaucetPrivateKeyExportModal" :faucetId="faucetId" />
        <v-card class="elevation-0">
            <v-alert text type="error" v-if="errorMessage">{{ errorMessage }}</v-alert>
            <v-card-text class="font-weight-medium error--text">
                <v-row>
                    Export your key if you need to access your faucet wallet. Make sure to do it before deleting the faucet.
                    <v-spacer></v-spacer>
                    <v-btn small depressed color="error" class="mt-2" @click="openExplorerFaucetPrivateKeyExportModal()"><v-icon class="mr-1">mdi-export</v-icon>Export Private Key</v-btn>
                </v-row>
                <v-divider class="my-5"></v-divider>
                <v-row>
                    Deleting the faucet is irreversible, if you haven't exported the private key before, funds will be lost.
                    <v-spacer></v-spacer>
                    <v-btn :loading="loading" small depressed color="error" class="mt-2" @click="deleteFaucet()"><v-icon class="mr-1">mdi-delete</v-icon>Delete Faucet</v-btn>
                </v-row>
            </v-card-text>
        </v-card>
    </v-sheet>
</template>

<script>
import { mapGetters } from 'vuex';
import ExplorerFaucetPrivateKeyExportModal from './ExplorerFaucetPrivateKeyExportModal';

export default {
    name: 'ExplorerFaucetSettingsDangerZone',
    props: ['faucetId'],
    components: {
        ExplorerFaucetPrivateKeyExportModal
    },
    data: () => ({
        errorMessage: null,
        loading: false,
    }),
    methods: {
        openExplorerFaucetPrivateKeyExportModal() {
            this.$refs.explorerFaucetPrivateKeyExportModal.open({ faucetId: this.faucetId });
        },
        deleteFaucet() {
            this.loading = true;
            const message = `Make sure you exported your private key before deleting the faucet. If you didn't, all funds stored on this address will be lost.

Deleting the faucet will:
- Delete drip history for this address.
- Delete the encrypted private key & all associated faucet parameters.

It won't be possible to restore the faucet afterwards. Do you still want to proceed?
            `;

            if (!confirm(message))
                return this.loading = false;

            this.server.deleteFaucet(this.faucetId)
                .then(() => this.$emit('delete'))
                .catch(console.log);
        }
    },
    computed: {
        ...mapGetters([
            'isBillingEnabled'
        ]),
    }
}
</script>
