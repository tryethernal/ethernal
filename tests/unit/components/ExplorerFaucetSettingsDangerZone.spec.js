import flushPromises from 'flush-promises';
import MockHelper from '../MockHelper';

import ExplorerFaucetSettingsDangerZone from '@/components/ExplorerFaucetSettingsDangerZone.vue';

beforeEach(() => jest.clearAllMocks());

describe('ExplorerFaucetSettingsDangerZone.vue', () => {
    let helper;
    const stubs = ['Explorer-Faucet-Private-Key-Export-Modal'];

    beforeEach(() => {
        helper = new MockHelper();
    });

    it('Should display danger zone', async () => {
        const wrapper = helper.mountFn(ExplorerFaucetSettingsDangerZone, { stubs });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });
});
