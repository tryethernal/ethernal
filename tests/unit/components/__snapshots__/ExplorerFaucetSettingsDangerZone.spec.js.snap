// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplorerFaucetSettingsDangerZone.vue Should display danger zone 1`] = `
<div class="v-sheet v-sheet--outlined theme--light rounded error">
  <explorer-faucet-private-key-export-modal-stub></explorer-faucet-private-key-export-modal-stub>
  <div class="elevation-0 v-card v-sheet theme--light">
    <!---->
    <div class="v-card__text font-weight-medium error--text">
      <div class="row">
        Export your key if you need to access your faucet wallet. Make sure to do it before deleting the faucet.
        <div class="spacer"></div> <button type="button" class="mt-2 v-btn v-btn--has-bg theme--light v-size--small error"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mr-1 mdi mdi-export theme--light"></i>Export Private Key</span></button>
      </div>
      <hr role="separator" aria-orientation="horizontal" class="my-5 v-divider theme--light">
      <div class="row">
        Deleting the faucet is irreversible, if you haven't exported the private key before, funds will be lost.
        <div class="spacer"></div> <button type="button" class="mt-2 v-btn v-btn--has-bg theme--light v-size--small error"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mr-1 mdi mdi-delete theme--light"></i>Delete Faucet</span></button>
      </div>
    </div>
  </div>
</div>
`;
